taf/oll/targetsdb.py

```diff
@@ -0,0 +1,149 @@
+import os
+from taf.oll.GitRepository import GitRepository
+from taf.oll.exceptions import TargetsNotFound
+from pathlib import Path
+
+# {
+#     'authentication_repo_name': {
+#         'commit' : {
+#             'target_path1': (target_git_repository1, custom_data1)
+#             'target_path2': (target_git_repository2, custom_data2)
+#             ...
+#         }
+#     }
+# }
+
+_targetsdb_dict = {}
+
+def load_targets(auth_repo, targets_classes=None, factory=None, root_dir=None, commit=None):
```

Comment:
I think we should not force a user to have a directory structure based on `targets` file.
It's fine to do that by default, but I suggest having another parameter (e.g. `path_mappings`), where we can pass something like:

```python
{
   'cityofsanmateo/law-xml': '/Users/daniel/xml-repos/smc-law',
   'cityofsanmateo/law-html': '/Users/daniel/html-repos/smc-law'
}
```

What are your thoughts?

Reply:
We have discussed and solved this. A factory function can be provided if using the default paths is not an option.